FE/src/app/todos/todo-details/todo-details.ts

```diff
@@ -1,11 +1,57 @@
-import { Component } from '@angular/core';
+import { ActivatedRoute, RouterModule, Router } from '@angular/router';
+import { CommonModule } from '@angular/common';
+import { TodoDataService } from '../todo-data-service';
+import { Todo } from '../../models/todo';
+import { Component, OnInit } from '@angular/core';
+
+import { MatCardModule } from '@angular/material/card';
+import { MatIconModule } from '@angular/material/icon';
+import { MatButtonModule } from '@angular/material/button';
+import { MatInputModule } from '@angular/material/input';
+import { MatSelectModule } from '@angular/material/select';
 
 @Component({
   selector: 'app-todo-details',
-  imports: [],
+  imports: [
+    RouterModule,
+    CommonModule,
+    MatCardModule,
+    MatIconModule,
+    MatButtonModule,
+    MatInputModule,
+    MatSelectModule,
+  ],
   templateUrl: './todo-details.html',
-  styleUrl: './todo-details.scss'
+  styleUrl: './todo-details.scss',
 })
-export class TodoDetails {
+export class TodoDetails implements OnInit {
+  todo: Todo | undefined;
+  constructor(
+    private todoDataService: TodoDataService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.activatedRoute.params.subscribe((params) => {
+      const id: number | undefined = Number(params['id']);
+      if (!id) {
+        this.router.navigate(['/todo-list']);
+      }
+      this.todoDataService.getTodo(id).subscribe((todo: Todo | undefined) => {
```

Comment:
This is not a recommended way of chaining multiple observables. It is better to use .pipe() with RxJs operators.
Probably, `mergeMap`, `switchMap`, `contactMap`, `exhaustMap` are operators your are looking for.